zeeguu_api/api/topics.py

```diff
@@ -0,0 +1,179 @@
+import flask
+import zeeguu
+from flask import request
+from zeeguu.model import Topic, TopicSubscription, TopicFilter
+
+from .utils.route_wrappers import cross_domain, with_session
+from .utils.json_result import json_result
+from . import api
+
+session = zeeguu.db.session
+
+SUBSCRIBE_TOPIC = "subscribe_topic"
+UNSUBSCRIBE_TOPIC = "unsubscribe_topic"
+SUBSCRIBED_TOPICS = "subscribed_topics"
+UNFOLLOWED_TOPICS = "unfollowed_topics"
+FILTER_TOPIC = "filter_topic"
+UNFILTER_TOPIC = "unfilter_topic"
+FILTERED_TOPICS = "filtered_topics"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{SUBSCRIBE_TOPIC}", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def subscribe_to_topic_with_id():
+    """
+    :param: topic_id -- the id of the topic to be subscribed to.
+    Subscribe to the topic with the given id
+
+    :return: "OK" in case of success
+    """
+
+    topic_id = int(request.form.get('topic_id', ''))
+
+    topic_object = Topic.find_by_id(topic_id)
+    TopicSubscription.find_or_create(session, flask.g.user, topic_object)
+
+    return "OK"
+
+
+# ---------------------------------------------------------------------------
+@api.route(f"/{UNSUBSCRIBE_TOPIC}/<topic_id>", methods=("GET",))
+# ---------------------------------------------------------------------------
```

Comment:
not clear subscribe is a POST and unsubscribe is a GET...
why? 
let's think well and stick to one!

Reply:
I agree, weird convention also used in feeds, will change that later this afternoon.